ndsl/stencils/testing/serialbox_to_netcdf.py

```diff
@@ -173,22 +185,53 @@ def main(
                         data_vars[varname] = get_data(
                             data_shape, total_ranks, n_savepoints, rank_list, varname
                         )
+                elif colapse_all_ranks:
+                    data_vars[varname] = get_data_collapse_all_ranks(
+                        total_ranks, n_savepoints, rank_list, varname
+                    )
                 else:
                     data_vars[varname] = get_data(
                         data_shape, total_ranks, n_savepoints, rank_list, varname
                     )
                 if len(data_shape) > 2:
                     encoding[varname] = {"zlib": True, "complevel": 1}
+
             dataset = xr.Dataset(data_vars=data_vars)
             dataset.to_netcdf(
                 os.path.join(output_path, f"{savepoint_name}.nc"), encoding=encoding
             )
 
 
+def get_data_collapse_all_ranks(total_ranks, n_savepoints, output_list, varname):
+    if total_ranks <= 0:
```

Comment:
`<= 0`? Can we have negative total_ranks??

Reply:
Probably not but I did a bot of "defensive coding" here. If we pass -1 for any reason we don't crash